runtime/include/spinlock.hpp

```diff
@@ -0,0 +1,63 @@
+#ifndef SPINLOCK_HPP
+#define SPINLOCK_HPP
+
+#include <atomic>
+#include <cstddef>
+#include <ctime>
+#include <time.h>
+
+class Spinlock {
+    std::atomic<unsigned int> lock;
+public:
+
+    // Default constructor to initialize the atomic_flag
+    Spinlock() : lock(0) {}
+
+    // Deleted copy constructor to prevent copying
+    Spinlock(const Spinlock&) = delete;
+    Spinlock& operator=(const Spinlock&) = delete;
+
+    Spinlock(Spinlock&& other) noexcept : lock() {
+        if (other.lock.exchange(1, std::memory_order_acquire)) {
+            lock.store(0, std::memory_order_release);
+        }
+    }
+
+    Spinlock& operator=(Spinlock&& other) noexcept {
+        if (this != &other) {
+            if (other.lock.exchange(1, std::memory_order_acquire)) {
+                lock.store(0, std::memory_order_release);
+            }
+        }
+        return *this;
+    }
+
+    void spin_lock() {
+        for(int i = 0; lock.load(std::memory_order_relaxed) || lock.exchange(1, std::memory_order_acquire); ++i) {
+            if(i==8){
+                i=0;
+                #if __APPLE__ 
+                __asm__ __volatile__ ("yield");
+                #else 
+                __asm__ __volatile__ ("pause")
```

Comment:
Why calling pause when __APPLE__ is not defined?

Reply:
forgot that this is only for APPLE, will remove this guard, thanks.